src/t_zset.c

```diff
@@ -2010,7 +2010,7 @@ void zremrangeGenericCommand(client *c, zrange_type rangetype) {
         }
     } else if (zobj->encoding == OBJ_ENCODING_SKIPLIST) {
         zset *zs = zobj->ptr;
-        dictDisallowResize(zs->dict);
+        dictPauseAutoResize(zs->dict);
```

Comment:
don't we want that in many other places, like ZPOP, SPOP, ZREM, SREM, HDEL etc?

Reply:
I suggested only doing it for commands likely to remove more than 90% of the fields, like removing a range.

I don't know if that's a good heuristic. Maybe it's better to do it everywhere where we did shrink before. I'm not sure.

Reply:
I think it's hard to say whether pausing auto-resizing in range deletion is better, because we may insert elements  after deletion immediately. The reason that I pause auto-resizing here is mainly compatible with the previous version. IMHO, I think shrinking should be an internal action and avoid being called by users as much as possible.